app/controllers/users_controller.rb

```diff
@@ -1,2 +1,114 @@
 class UsersController < ApplicationController
-end
+
+  before_action :find_user, only: [:show]
+
+  # TBD - What are we doing for  login => Authentication or Authorization?
+
+  # def login_form
+  #   @user = User.new
+  # end
+  # ==> Do we need a login_form html.erb file?
+  def create
+    @user = User.find_by(username: params[:username])
+
+    if guest?
+      @user = guest_user
+      session[:user_id] = @user.id
+      session[:guest_user_id] = nil
+      redirect_to user_path(@user)
+    elsif @user == nil
+      flash[:error] = 'Username does not exist'
+      redirect_to login_path
+    else
+      if @user.authenticate(params[:password])
+        session[:user_id] = @user.id
+        redirect_to user_path(@user)
+      else
+        flash[:error] = 'Username and Password do not match'
+        redirect_to login_path
+      end
+    end
+  end
+
+  def login
+    user = User.find_by(username: params[:user][:username])
+    # New User
+    if user.nil?
+      user = User.create(username: params[:user][:username])
+      unless user.save
+        flash[:error] = 'Unexpected error occured. Login unsuccessful.'
+        redirect_to root_path
+        return
+      end
+      flash[:welcome] = "Welcome #{user.username}"
+    else
+      # Existing User
+      flash[:welcome] = "Welcome back #{user.username}"
+    end
+
+    session[:user_id] = user.id
+    redirect_to root_path
+  end
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    if @user.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def current
+    @current_user = User.find_by(id: session[:user_id])
+    unless @current_user
+      flash[:error] = 'You must be logged in to see this page'
+      redirect_to root_path
+      return
+    end
+  end
+
+  def logout
```

Comment:
I'm sure there are multiple ways to approach this but this could change to a simple destroy action, similar to the OAuth reading.